loadtest/tests.go

```diff
@@ -290,6 +291,135 @@ func createPost(c *EntityConfig, team *UserTeamImportData, channelId string) {
 	}
 }
 
+func createChannel(c *EntityConfig, teamId string, userId string) (*model.Channel, error) {
+	const (
+		PUBLIC_CHANNEL = iota + 1
+		PRIVATE_CHANNEL
+		DIRECT_CHANNEL
+		GROUP_CHANNEL
+	)
+
+	choices := []randutil.Choice{
+		{
+			Weight: int(c.LoadTestConfig.UserEntitiesConfiguration.PublicChannelCreationChance * 1000),
+			Item:   PUBLIC_CHANNEL,
+		},
+		{
+			Weight: int(c.LoadTestConfig.UserEntitiesConfiguration.PrivateChannelCreationChance * 1000),
+			Item:   PRIVATE_CHANNEL,
+		},
+		{
+			Weight: int(c.LoadTestConfig.UserEntitiesConfiguration.DirectChannelCreationChance * 1000),
+			Item:   DIRECT_CHANNEL,
+		},
+		{
+			Weight: int(c.LoadTestConfig.UserEntitiesConfiguration.GroupChannelCreationChance * 1000),
+			Item:   GROUP_CHANNEL,
+		},
+	}
+
+	choice, err := randutil.WeightedChoice(c.r, choices)
+	if err != nil {
+		return nil, err
+	}
+
+	if choice.Item == PUBLIC_CHANNEL || choice.Item == PRIVATE_CHANNEL {
+		channel := &model.Channel{}
+		channel.TeamId = teamId
+		channel.Name = makeChannelName(c.r.Int())
+		channel.DisplayName = makeChannelDisplayName(c.r.Int())
+
+		if choice.Item == PUBLIC_CHANNEL {
+			channel.Type = "O"
+		} else {
+			channel.Type = "P"
+		}
+
+		newChannel, resp := c.Client.CreateChannel(channel)
+		if resp.Error != nil {
+			return nil, resp.Error
+		}
+
+		return newChannel, nil
+	} else if choice.Item == DIRECT_CHANNEL || choice.Item == GROUP_CHANNEL {
+
+		users, resp := c.Client.GetUsers(0, 100, "")
```

Comment:
If I follow this correctly, this will limit us to a maximum of 10,000 direct message channels. In practice, DMs account for 95+% of the channels in a production table once an installation hits 20,000+ users.

This also incurs an extra "GetUsers" cost that isn't technically part of the action of creating a new channel, and thus doesn't simulate a user entity very well.

Could we leverage our pre-existing knowledge of the users on the system to drive this instead?

Reply:
Yes, you are right. This is not the first time that I encounter this issue and it's going to be addressed in a brief review document I am working on (and will publish soon), essentially expressing my thoughts as a completely new user/developer approaching this codebase.

Coming back to this PR. Is having all the users loaded in memory before starting the tests and randomly picking some the best way to simulate this case?  

I am thinking that maybe a `/users/search` call might be more close to reality. That obviously poses different problems as the user needs to know what to look for.  
Maybe the two approaches could be used in conjunction. We pick a user randomly from the in memory state and then we have the user entity search for him.

These are just some thoughts, please let me know what you think.

Reply:
Agreed with both your ideas: have a pre-existing knowledge of users, and then drive a few queries to `/users/search` like a human would when constructing such a channel.